include/fmt/format.h

```diff
@@ -3563,19 +3557,31 @@ inline format_to_n_result<OutputIt> vformat_to_n(
  end of the output range.
  \endrst
  */
-template <typename OutputIt, typename... Args>
-inline format_to_n_result<OutputIt> format_to_n(
-    OutputIt out, std::size_t n, string_view format_str, const Args &... args) {
-  return vformat_to_n<OutputIt>(
-    out, n, format_str, make_format_to_n_args<OutputIt>(args...));
+template <typename OutputIt, typename String, typename... Args>
+inline typename std::enable_if<
+  internal::is_format_string<String>::value &&
+  std::is_same<FMT_CHAR(String), char>::value,
+  format_to_n_result<OutputIt>>::type format_to_n(
+    OutputIt out, std::size_t n, const String &format_str, const Args &... args) {
+  internal::check_format_string<Args...>(format_str);
+  typedef FMT_CHAR(String) char_t;
+  typedef format_to_n_context<OutputIt> context_t;
+  format_arg_store<context_t, Args...> as{ args... };
+  return vformat_to_n<OutputIt>(out, n, basic_string_view<char_t>(format_str), as);
 }
-template <typename OutputIt, typename... Args>
-inline format_to_n_result<OutputIt> format_to_n(
-    OutputIt out, std::size_t n, wstring_view format_str,
-    const Args &... args) {
+
+template <typename OutputIt, typename String, typename... Args>
+inline typename std::enable_if<
+  internal::is_format_string<String>::value &&
+  !std::is_same<FMT_CHAR(String), char>::value,
+  format_to_n_result<OutputIt>>::type format_to_n(
+    OutputIt out, std::size_t n, const String &format_str, const Args &... args) {
```

Comment:
Why do we need two overloads?

Reply:
Well, `format_to_n` in `master` has different implementations in the function body depending on the actual char type. Therefore I mapped one overload (resembling the old `string_view` one) to the `char` case, and the other (resembling the old `wstring_view` one) to all other `char` types. The `enable_if` switches between the two. At this stage, I haven't dug into the different iterator types used in the function bodies to figure out if they can be merged - opening the road to remove this `enable_if` switch.

Reply:
Whoa, the widened but constrained interface triggers a spectacular failure on all gccs and clangs! I suspect a weird interaction with their stl implementation. MSVC is just fine ...

Reply:
I think the difference between the two is incidental (I just forgot to update the `wchar_t` overload after changing the `char` one). Both should forward to `vformat_to_n` and can be merged.